### PR TITLE
Add support for DECRQM/DECRPM

### LINKED ISSUE #100: Add support for DECRQM
It's used to report whether mode is supported/set, etc.

This is used by sync updates as a way to detect the feature support. See https://gist.github.com/christianparpart/d8a62cc1ab659194337d73e399004036?permalink_comment_id=3946967#feature-detection

The mode could still be used without it, but it probably worth adding support for this escape in general.

### PR DESCRIPTION
**Pull request by kchibisov**
Fixes #100.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 CHANGELOG
 =========
 
+## Unreleased
+
+- Implement `Clone + Copy` for `ansi::Mode
+- Add `Handler::write\_payload` to forward data from parser back to PTY
+- Add `Handler::report\_mode` handling DECRQM/DECRPM
```
Comment (chrisduerr) [inline CHANGELOG.md:8]: This is not what the function is called.

File: src/ansi.rs [hunk 2]
```diff
@@ -660,6 +667,9 @@ pub trait Handler {
     ///
     /// The output is of form `CSI > 4 ; mode m`.
     fn report_modify_other_keys(&mut self) {}
+
+    /// The VTE handler requested to write payload to the terminal.
+    fn pty_write(&mut self, _payload: String) {}
```
Comment (chrisduerr) [inline src/ansi.rs:672]: VTE writing arbitrary stuff shouldn't be a thing. There's a reason why this is a trait-based interface, we should have proper functions here.

File: src/ansi.rs [hunk 3]
```diff
@@ -577,6 +581,9 @@ pub trait Handler {
     /// Set mode.
     fn set_mode(&mut self, _mode: Mode) {}
 
+    /// DECRPM - Report terminal mode.
+    fn report_mode(&mut self, _reporter: ModeReporter, _mode: Mode) {}
```
Comment (chrisduerr) [inline src/ansi.rs:585]: Why is this passing two separate modes?

**Review by chrisduerr — CHANGES_REQUESTED** [review]

**Comment by kchibisov** [comment]
I guess this is fine? Or do we want a similar `Reporter` thing I had to help building downstream an escape sequence?

**Review by chrisduerr — APPROVED** [review]

**Comment by kchibisov** [comment]
Added `Clone + Copy` on private mode, because it was on public, but not on private one.

**Pull request merged**